module/localisation/RobotParticleLocalisation/src/RobotParticleLocalisation.cpp

```diff
@@ -123,6 +129,36 @@ namespace localisation {
                     }
                 }
             });
+
+        on<Trigger<ResetRobotHypotheses>, With<Sensors>, Sync<RobotParticleLocalisation>>().then(
+            "Reset Robot Hypotheses", [this](const ResetRobotHypotheses& locReset, const Sensors& sensors) {
+
+                Transform3D Hft;
+                Transform3D Hfw;
+                const Transform3D& Htw = convert<double, 4, 4>(sensors.world);
+                arma::vec3 rTFf;
+                arma::mat22 Hfw_xy;
+                arma::mat22 pos_cov;
+                arma::mat33 state_cov;
+                std::vector<arma::vec3> states;
+                std::vector<arma::mat33> cov;
+
+                for (auto& s : locReset.hypotheses) {
+                    rTFf              = {s.position[0], s.position[1], 0};
```

Comment:
initialise these variables in the for loop. cpp is smart enough to optimise and not re-create it every loop. this also removes the possibility of using values from the last loop